firedrake/slate/slac/optimise.py

```diff
@@ -235,6 +236,8 @@ def _drop_double_transpose_transpose(expr, self):
     if isinstance(child, Transpose):
         grandchild, = child.children
         return self(grandchild)
+    elif child.terminal and child.rank > 1:
+        return Tensor(adjoint(child.form))
```

Comment:
Okay. I changed this so that it is dealt with in Slate's optimiser. How is that? Now all tests are passing :)

Reply:
OK, I think this works.